### ISSUE TITLE
Working?

### ISSUE THREAD

**Issue by msoldin** [Silveryu/vhdl-neuralnet#1]
Hey, i was not able to message you. @Silveryu  
Can i ask some questions maybe? 

So, did you ran into some problems regarding fpga place? Because my solution is nearly the same with the matrix multiplication.

And was your achievement worth the work?

Kindly regards,
m.soldin

**Comment by Silveryu** [comment]
Hey, sure thing.

Let's call what's on GitHub the non-pipeline version (our final project was pipelined but it was lost, unfortunately). In our project, we found the whole process took more than one FPGA clock cycle so we had to pipeline the whole thing, at which point using an FPGA for this purpose is a bit redundant (it's just working as a glorified GPU). This was due to the size of what we were processing, we tried (as is in the pdf presentation) to do pooling on the MNIST data to reduce it but it also wasn't enough ( another problem with pooling was that the accuracy dropped considerably, so it won't be that impressive a project )
I don't know if your project also involves MNIST (we found the idea of it recognizing an image and showing it on the displays appealing ). But I think you'd be better off with just a small example dataset (that way no pipeline would be needed and the non-pipeline version present here is appropriate). I thought the https://archive.ics.uci.edu/ml/datasets/student+performance# dataset, where you predict a student's grade, was also a good (and more importantly, easier) idea.

Good luck!

**Comment by msoldin** [comment]
Well thanks for this answer!
At the moment we try to solve a very simple problem (XOR as Neural Network) with quantized weights. (we started the implementation one week ago, so were just at the beginning)
We're using the PYNQ-Z1 Board with around 220 DSP Slices. So the question was, is this enough for computation. Because every multiplication is parallel, taking one of these slices.

I guess we will see and optimize it a little bit (Maybe with pipe lining). After the XOR we also planned to use a MNIST-Dataset. If you want to (just for comparing), i can send you our achievements at the end of the project :)

But thank you very much for the answer, i appreciate that.

**Comment by Silveryu** [comment]
Glad to help, i agree it'd be interesting to compare results later, as the project didn't quite meet my initial expectations (would be interesting to do backprop for example). Godspeed!
